src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -122,6 +123,7 @@ public function boot()
         }
 
         $this->booted = true;
+        $this->httpKernel = $this->getHttpKernel();
```

Comment:
this looks wrong to be. It is perfectly possible to boot the kernel without needing the HttpKernel (perfect use case is the console)